source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRViewController.mm

```diff
@@ -225,20 +226,18 @@ - (void) addTasksToConcurrentQueue:(std::vector<std::shared_ptr<BaseCardElement>
                 [self tagBaseCardElement:elem];
                 /// dispatch to concurrent queue
                 std::shared_ptr<TextBlock> txtElem = std::dynamic_pointer_cast<TextBlock>(elem);
-                ACOHostConfig *acoConfig = [[ACOHostConfig alloc] initWithConfig:_hostConfig];
                 dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0),
                     ^{
                         std::string dateParsedString = [ACOHostConfig getLocalizedDate:txtElem];
                         // MarkDownParser transforms text with MarkDown to a html string
                         std::shared_ptr<MarkDownParser> markDownParser = std::make_shared<MarkDownParser>(dateParsedString.c_str());
                         NSString *parsedString = [NSString stringWithCString:markDownParser->TransformToHtml().c_str() encoding:NSUTF8StringEncoding];
-
                         // Font and text size are applied as CSS style by appending it to the html string
-                        NSString *fontFamily = [NSString stringWithCString:_hostConfig->fontFamily.c_str() encoding:NSUTF8StringEncoding];
-                        const int fontWeight = [acoConfig getTextBlockFontWeight:txtElem->GetTextWeight()];
-                        parsedString = [parsedString stringByAppendingString:[NSString stringWithFormat:@"<style>body{font-family: '%@'; font-size:%dpx; font-weight: %d;}</style>",
-                                                                              fontFamily,
-                                                                              [acoConfig getTextBlockTextSize:txtElem->GetTextSize()],
+                        const int fontWeight = [_hostConfig getTextBlockFontWeight:txtElem->GetTextWeight()];
+                        parsedString = [parsedString stringByAppendingString:[NSString stringWithFormat:@"<style>body{font-family: '%@', '%@'; font-size:%dpx; font-weight: %d;}</style>",
+                                                                              _hostConfig.fontFamilyNames[0],
+                                                                              _hostConfig.fontFamilyNames[1],
```

Comment:
This only works if there are exactly two entries in the array, which might not be the case.

Reply:
thanks

---
In reply to: [172342962](https://github.com/Microsoft/AdaptiveCards/pull/1207#discussion_r172342962) [](ancestors = 172342962)